cms/forms/wizards.py

```diff
@@ -158,15 +158,23 @@ def create_page_titles(page, title, languages):
                 create_title(language, title, page)
 
     @staticmethod
-    def get_first_placeholder(page):
+    def get_placeholder(page, slot=None):
         """
-        Returns the first editable, non-static placeholder or None.
+        Returns the named placeholder or the first editable, non-static
+        placeholder or None.
         """
-        for placeholder in page.get_placeholders():
-            if not placeholder.is_static and placeholder.is_editable:
-                return placeholder
-        else:
-            return None
+        placeholders = page.get_placeholders()
+
+        if slot:
```

Comment:
I think it might be better to do a single loop and tackle the slot condition from there